src/components/RequesterLinks/FullNameLink.js

```diff
@@ -0,0 +1,27 @@
+import { Link } from 'react-router-dom';
+import { FormattedMessage } from 'react-intl';
+
+import { getFullName } from '../../utils';
+import propTypes from './propTypes';
+
+// This link component takes both the userId and the user object
```

Comment:
We can improve comment. Just for example:
/**
 * FullNameLink component displays a link to the user's profile using their userId and user object.
 * - If the user object is present, it renders a link with the user's full name.
 * - If the user is anonymized (userId === null), it displays an anonymized message.
 * - If the user is not found, it displays an unknown user message.
 *
 * @param {Object} props
 * @param {string|null|undefined} props.userId - The user's identifier.
 * @param {Object|null|undefined} props.user - The user object.
 * @returns {JSX.Element} A link to the user profile or a message.
 */